cmd/operator/main.go

```diff
@@ -183,6 +185,13 @@ func run(fs *flag.FlagSet) int {
 		stdlog.Fatal(err)
 	}
 
+	l := func(format string, a ...interface{}) {
```

Comment:
I think it should be behind a feature flag, wdyt?

Reply:
We still don't have feature flags tho 😅 

But I agree that this needs to be configurable, e.g. what's the side-effect of having gomaxprocs auto enabled during development? If I run the operator in my machine, will it take over all cores?

Reply:
Read the `CGroup` value in the container environment to identify the `CPU quota` of the container, calculate the actual number of cores, and automatically set the number of `GOMAXPROCS` threads.

Reply:
Well!
I think it's a fair and simple feature flag implementation.
I'm ok with that, because it's also the same approach we have in Prometheus.
Let's see what the other maintainers think about it.

Reply:
I have another PR open for a few weeks implementing feature-flags as well, but following Kubernetes feature-flags approach. https://github.com/prometheus-operator/prometheus-operator/pull/6491

Differently from Prometheus' approach, K8s allows us maintainers to graduate feature flags slowly while still letting users disable feature flags that we might enable by default in the future.

In my opinion, this gives us the flexibility to enable features when we feel it has achieved enough maturity, while also providing safe-guards to users if they break something in production